diesel/src/upsert/on_conflict_extension.rs

```diff
@@ -279,25 +277,26 @@ impl<Stmt, Target> IncompleteOnConflict<Stmt, Target> {
     ///     .do_update()
     ///     .set(name.eq("I DONT KNOW ANYMORE"))
     ///     .execute(conn);
+    /// # #[cfg(any(feature = "sqlite", feature = "postgres"))]
     /// assert_eq!(Ok(1), insert_count);
+    /// # #[cfg(feature = "mysql")]
+    /// assert_eq!(Ok(2), insert_count);
```

Comment:
This was a little funky to me but I found documentation at the MariaDB website [here](https://mariadb.com/kb/en/insert-on-duplicate-key-update/#:~:text=ON%20DUPLICATE%20KEY%20UPDATE%20is,API's%20CLIENT_FOUND_ROWS%20flag%20is%20set.)

> The row/s affected value is reported as 1 if a row is inserted, and 2 if a row is updated, unless the API's CLIENT_FOUND_ROWS flag is set.

A little bit odd to me, so happy to either see if we can set this flag somewhere or perhaps just document this behavior everywhere in the tests?

Reply:
This flag already set here: https://github.com/diesel-rs/diesel/blob/983209a6188392552efa51fe76284eb3097b012a/diesel/src/mysql/connection/url.rs#L121

Reply:
🤔  maybe this isn't set in the doctest setup then; will look again!

Reply:
Oh no... it totally is set in the doctests setup... I think I'm just misunderstanding something

> For [INSERT ... ON DUPLICATE KEY UPDATE](https://dev.mysql.com/doc/refman/8.0/en/insert-on-duplicate.html) statements, the affected-rows value per row is 1 if the row is inserted as a new row, 2 if an existing row is updated, and 0 if an existing row is set to its current values. If you specify the CLIENT_FOUND_ROWS flag, the affected-rows value is 1 (not 0) if an existing row is set to its current values.